assets/js/modules/ads/datastore/moduleData.test.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * `modules/ads` moduleData store tests.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	createTestRegistry,
+	untilResolved,
+	unsubscribeFromAll,
+} from '../../../../../tests/js/utils';
+import { initialState } from './index';
+import { MODULES_ADS } from './constants';
+
+describe( 'modules/ads module data', () => {
+	const baseModulesDataVar = '_googlesitekitModulesData';
+	const baseData = {
+		ads: {
+			supportedConversionEvents: [ 'add-to-cart' ],
+		},
+	};
+
+	let registry;
+
+	beforeEach( () => {
+		global[ baseModulesDataVar ] = baseData;
+		registry = createTestRegistry();
+	} );
+
+	afterEach( () => {
+		delete global[ baseModulesDataVar ];
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'receiveModuleData', () => {
+			it( 'requires the moduleData param', () => {
+				expect( () => {
+					registry.dispatch( MODULES_ADS ).receiveModuleData();
+				} ).toThrow( 'moduleData is required.' );
+			} );
+
+			it( 'receives module data', async () => {
+				await registry
+					.dispatch( MODULES_ADS )
+					.receiveModuleData( baseData );
+
+				expect(
+					registry.select( MODULES_ADS ).getModuleData()
+				).toEqual( { supportedConversionEvents: undefined } );
```

Comment:
Why is this `undefined`? From the `supportedConversionEvents` value in `baseData`, this should be `[ 'add-to-cart' ]`, right?

Reply:
@tofumatt good catch, I think I need to call the selector first, then resolve getModuleData and then select again, like the other tests. I think I can update this here and remove one of the other tests that check for this, or remove this one entirely as I check for the events further down. I'll amend.